src/Containers/Microsoft.NET.Build.Containers/Tasks/CreateNewImage.cs

```diff
@@ -235,6 +235,20 @@ private static void SetEnvironmentVariables(ImageBuilder img, ITaskItem[] envVar
     private void SafeLog(string message, params object[] formatParams) {
         if(BuildEngine != null) Log.LogMessage(MessageImportance.High, message, formatParams);
     }
+    private void SafeLog(LogMessage message) {
```

Comment:
minor: consider pattern matching.

https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression

Reply:
similar to the switch expression state machine in Registry.cs, switch expressions cannot return 'void' in the C# language right now. So I could either do the same immediately-executed-Action hack, or use the old-style switch here.